web/index.html

```diff
@@ -552,64 +601,147 @@ <h3>Used in</h3>
                     };
 
                     const animateScroll = (currentTime) => {
+                        if (this.isScrolling) {
+                            return;
+                        }
                         const timeElapsed = currentTime - startTime;
-                        const run = easeInOutQuad(timeElapsed, start, position - start, duration);
-
+                        const run = easeInOutQuad(timeElapsed, start, 0 - start, duration);
                         window.scrollTo(0, run);
-
                         if (timeElapsed < duration) requestAnimationFrame(animateScroll);
                     };
 
                     requestAnimationFrame(animateScroll);
                 },
                 clearSaveData: function () {
-                    this.saveData = [];
+                    this.saveData = new Set();
+                    this.saveDataArray = [];
                     localStorage.setItem("saveData", "");
+                    window.location.reload();
+                },
+                dragStart() {
+                    this.isDraggingBookmarklet = true;
+                },
+                dragEnd() {
+                    this.isDraggingBookmarklet = false;
+                },
+                preventClick(event) {
+                    this.isShaking = true;
+                    setTimeout(() => {
+                        this.isShaking = false;
+                    }, 250);
+                    event.preventDefault();
                 },
                 formatNumber: function (n) {
                     if (n === undefined || n <= 0) {
                         return "#####"
                     }
                     return n.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
                 },
-                loadChunk: async function (chunk) {
-                    return fetch(DATA_PREFIX + `chunks/chunk-${chunk}.json`)
-                        .then(response => {
-                            console.log("Loaded chunk", chunk);
-                            return response.json()
-                        })
-                        .then(data => {
-                            let [sep1, sep2, sep3] = [":3", ":2", ":1"]
-                            chunkData = Object.fromEntries(Object.entries(data).map(x => {
-                                let [from, to, hiddenFrom, hiddenTo] = x[1].split(sep1)
-                                if (from == "") {
-                                    from = []
-                                } else {
-                                    from = from.split(sep2).map(x => x.split(sep3))
```

Comment:
no idea why github is trying to insert the iscraftingtree method halfway through the old chunkload method, but sorry that it makes the next part more unreadable lol